Sources/TuistGenerator/Generator/ProjectGroups.swift

```diff
@@ -115,4 +120,14 @@ class ProjectGroups {
         }
         return groupNames
     }
+    
+    private static func extractProjectTextSettings(from options: [Project.Options]) -> Project.Options.TextSettings? {
```

Comment:
What about adding this to an extension of Array where the `Element` is `ProjectOptions`?

```swift
// ProjectOptions.swift

extension Array where Element == ProjectOptions {
  func textSettings() -> TextSettings? {
  
  }
}
```
That way we can unit-test this method in isolation.